src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -152,105 +153,64 @@ public async Task<IReadOnlyCollection<JobInfo>> DequeueJobsAsync(byte queueType,
+            sqlCommand.Parameters.AddWithValue("@QueueType", queueType);
+            sqlCommand.Parameters.AddWithValue("@Worker", worker);
+            sqlCommand.Parameters.AddWithValue("@HeartbeatTimeoutSec", heartbeatTimeoutSec);
+            if (jobId.HasValue)
             {
-                try
-                {
-                    using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true);
-                    using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand();
-
-                    // cannot use VLatest as it incorrectly asks for optional @InputJobId
-                    sqlCommandWrapper.CommandText = "dbo.DequeueJob";
-                    sqlCommandWrapper.CommandType = System.Data.CommandType.StoredProcedure;
-                    sqlCommandWrapper.Parameters.AddWithValue("@QueueType", queueType);
-                    sqlCommandWrapper.Parameters.AddWithValue("@Worker", worker);
-                    sqlCommandWrapper.Parameters.AddWithValue("@HeartbeatTimeoutSec", heartbeatTimeoutSec);
-                    if (jobId.HasValue)
-                    {
-                        sqlCommandWrapper.Parameters.AddWithValue("@InputJobId", jobId.Value);
-                    }
-
-                    await using SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(cancellationToken);
-                    JobInfo jobInfo = await sqlDataReader.ReadJobInfoAsync(cancellationToken);
-                    if (jobInfo != null)
-                    {
-                        jobInfo.QueueType = queueType;
-                    }
-
-                    return jobInfo;
-                }
-                catch (Exception ex)
-                {
-                    attempt++;
-                    bool isRetriable = ex.IsRetryable();
-
-                    _logger.LogError(ex, "DequeueAsync failed. Attempt {AttemptNumber}. Is retriable: {IsRetriable}", attempt, isRetriable);
-
-                    if (!isRetriable)
-                    {
-                        throw;
-                    }
+                sqlCommand.Parameters.AddWithValue("@InputJobId", jobId.Value);
+            }
 
-                    if (attempt >= MaxNumberOfAttempts)
-                    {
-                        if (isRetriable)
-                        {
-                            throw new RetriableJobException(ex.Message, ex);
-                        }
-                        else
-                        {
-                            throw;
-                        }
-                    }
-                }
+            JobInfo jobInfo = await _sqlRetryService.ExecuteSqlDataReaderFirstRow(
+                sqlCommand,
+                JobInfoExtensions.LoadJobInfo,
+                _logger,
+                "DequeueAsync failed.",
+                cancellationToken);
```

Comment:
This is removing previously added behavior of retrying because of dequeue conflicts

Reply:
Retries are now inside _sqlRetryService.ExecuteSqlDataReaderFirstRow.